### DISCUSSION TITLE
Force Sync Logs in Pino to avoid "empty" errors in logs

### DISCUSSION THREAD

**Discussion by gabrsar** [directus/directus#14012] [Archive]
Hi everyone.

I'm having some issues where my logs inside async functions are coming with errors as empty objects (`{}`):

For example:

```typescript
export default defineHook(({ schedule, action }, { logger, database, env }) => {
  const myService = new MyService(logger, database, env);

  action("orders.items.update", async (event) => {
    const result = await myService.doSomethingAsync();
    return result.status;
  });
});

class MyService {
  constructor(logger: Logger, database: Knex, env: any) {
    this.logger = logger;
    this.database = database;
    this.env = env;
  }

  async doSomethingAsync() {
    try {
      // do something...
    } catch (error) {
      this.logger.error({ error }, "Something bad happened 1"); // Attempt 1
      this.logger.error({ error: JSON.stringify(error) }, "Something bad happened 2 "); // Attempt 2
    }
  }
}
```

And my logs are showing something like: 
```json
    {"level":50,"error":{},"message":"Something bad happened 1"}
    {"level":50,"error":"{}","message":"Something bad happened 2"}
```

Same problem:
https://stackoverflow.com/questions/57758973/pino-error-log-is-empty-although-error-object-contains-information

Cause:
https://github.com/pinojs/pino/blob/HEAD/docs/asynchronous.md

I could find how to change it on directus through configs looking on:
https://github.com/directus/directus/blob/ce0169d5c83f79449e5b0b69836b8b9aa809a0bf/api/src/logger.ts

Anyone has this kind of problem too? Any way to configure the destination of the Pino logger through the env vars of directus?

Thank you very much.

**Comment by rijkvanzanten** [comment]
Great question. I don't think this is something that can currently be configured through the env vars alone. In order to enable the `sync` render of the logs, it looks like the pino options have to be wrapped in another `pino.destination()` call, which is currently not implemented. 

Have you been seeing the same error problem in "regular" Directus usage (outside your extension)? If so, it might be worth tracking this as a bug instead 🙂

**Comment by gabrsar** [comment]
@rijkvanzanten, no, I only saw this issue in my own logs so far.

**Reply by rijkvanzanten** [reply]
Interesting! Based on the description and linked docs, I'd assume the same problem should exist for regular Directus errors 🤔 @br41nslug could you take a look?
